support/hololens/ServoApp/ServoControl/ServoControl.cpp

```diff
@@ -89,6 +89,48 @@ void ServoControl::OnLoaded(IInspectable const &, RoutedEventArgs const &) {
   StartRenderLoop();
 }
 
+void ServoControl::InitializeTextController() {
+  mInputPane = Windows::UI::ViewManagement::InputPane::GetForCurrentView();
+  auto manager = CoreTextServicesManager::GetForCurrentView();
+  mEditContext = manager.CreateEditContext();
+  mEditContext->InputPaneDisplayPolicy(CoreTextInputPaneDisplayPolicy::Manual);
+  mEditContext->InputScope(CoreTextInputScope::Text);
+  mEditContext->TextRequested([=](const auto &, const auto &e) {
+    e.Request().Text(L"");
+    CoreTextRange sel;
+    sel.StartCaretPosition = 0;
+    sel.EndCaretPosition = 0;
+    e.Request().Range() = sel;
+  });
+
+  mEditContext->TextUpdating([=](const auto &, const auto &e) {
+    auto keystr = *hstring2char((e.Text()));
+    RunOnGLThread([=] { mServo->KeyDown(keystr); });
```

Comment:
This is scary due to #27115. We should dereference keystr inside of the closure, rather than storing a pointer to the inner string value and using that.